components/linking_landing_page/index.tsx

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import LinkingLandingPage from './linking_landing_page';
+
+function mapStateToProps(state: GlobalState) {
+    const config = getConfig(state);
+    const defaultTheme = getTheme(state);
+    const location = window.location.href.replace('/vault#', '');
```

Comment:
I know this probably isn't a big deal since this component is meant to live at this URL but this replace is tightly coupling this index to the URL path it's being served at. If we were to move it to a different path this would break. Is there a way to do this replace differently?

Side note: Also feels wrong to be re-doing this replace every time mapStateToProps runs. Probably not a big deal since this page is likely very static.

Reply:
Agreed on not running the `replace` in `mapStateToProps`.

I suppose we could have this be a config item, and the route name can be configured by the server if necessary. Does that work?

Reply:
That could work but is pretty heavyweight. What about just grabbing the bits after the `#` and reconstructing the URL with siteURL pulled in from the config? Is that possible?

Reply:
I suppose so, but that sounds more complicated than it needs to be. I've moved the `replace` function to `componentDidMount()` so that it's only run once now, which should make it less CPU-heavy. I think that should suffice, what do you think?

Reply:
I'm good with that. Wasn't really worried too much about the CPU cost, more worried about the coupling to the URL but it's not a huge deal and it sounds like it's a pain to generalize. I'm good with it 👍